src/plugins/intel_gpu/src/runtime/ocl/ocl_memory.cpp

```diff
@@ -110,6 +110,13 @@ event::ptr gpu_buffer::copy_from(stream& stream, const void* host_ptr) {
     return ev;
 }
 
+event::ptr gpu_buffer::copy_to(stream& stream, void* host_ptr) {
+    auto& cl_stream = downcast<ocl_stream>(stream);
+    cl_stream.get_cl_queue().enqueueReadBuffer(_buffer, true, 0, size(), host_ptr, nullptr, nullptr);
```

Comment:
Why is it blocking call? You can create base event here and let the caller decide if wait is needed or not.